ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/NVidia_Cuda_Toolkit/tasks/main.yml

```diff
@@ -11,19 +11,19 @@
 # RedHat 7 and Ubuntu 16 on x86_64
 - name: Download NVidia CUDA toolkit
   get_url:
-    url="https://developer.nvidia.com/compute/cuda/9.0/Prod/local_installers/cuda_9.0.176_384.81_linux-run"
-    dest="/tmp/cuda9_linux-run"
+    url: https://developer.nvidia.com/compute/cuda/9.0/Prod/local_installers/cuda_9.0.176_384.81_linux-run
```

Comment:
Does OpenJ9 require this particular version?  The latest release as of 25-Jan-2018 is 9.1.85.
Same comment applies for different OSes listed below.

Reply:
The Nvidia toolkit is not OSS, and [the license](https://docs.nvidia.com/cuda/eula/index.html) should be reviewed for suitability in AdoptOpenJDK usage.  My layman reading of Section 1.2.1 (x) is of a field of use restriction that is incompatible with our outgoing license; *but* that requires a proper legal interpretation - **don't take my word for it**.

Reply:
No it does not. OpenJ9 requires [CUDA 7.5](https://developer.nvidia.com/cuda-75-downloads-archive).
Is there an issue with the [7.5 license](http://developer.download.nvidia.com/compute/cuda/7.5/Prod/docs/sidebar/EULA.pdf).
If there are no legal issues then I'll downgrade the CUDA version.

Reply:
OMR currently allows use of versions 5.5 through 9.0, inclusive.

My reading of the JIT code in GpuHelpers.cpp [1] is that it supports versions 5.5 through 7.5 on Windows; newer versions might work on Linux (via a symbolic link named libcudart.so), but I don't know how much testing has occurred. @gita-omr may have more information to share.

I suggest we install and build with version 7.5 which yields JDKs that will be more widely usable (they will work with a system which has a runtime version 7.5 through 9.0 installed - with the above caveat for JIT); if built with 9.0 they will require runtime version 9.0 (versions 7.5, 8.0 will not be accepted by OMR) and may not work for JIT.

[1] https://github.com/eclipse/openj9/blob/72c97f3db39b765967856190c3f306cd3cce7ee7/runtime/compiler/trj9/runtime/GPUHelpers.cpp#L369

Reply:
Another consideration is that CUDA 8.0 and newer *don't* support Visual Studio 2010 which is used to compile JDK8 on Windows. Using CUDA 7.5 for all platforms, at least for consistency, seems sensible.

Reply:
OMR now supports up to version 9.2. CUDA 9.2 is the only version that is compatible with newer versions of gcc (e.g. 7).

Reply:
@keithc-ca I'm concerned that your last two messages are now in conflict - you recommended 7.5, but that one won't work when we build with gcc7. Does this mean your recommendation is to use 9.2 or higher now?

Reply:
Given the need to support gcc7, I don't see much choice at this point. I think we need to use CUDA 9.2 (and fix any JIT issues that arise from that). @gita-omr Care to comment?